src/System.CommandLine/ParseResult.cs

```diff
@@ -180,12 +181,12 @@ public IReadOnlyList<string> UnmatchedTokens
                 }
             }
 
-            static T? Convert(ArgumentConversionResult validatedResult)
+            static T Convert(ArgumentConversionResult validatedResult)
             {
                 var convertedResult = validatedResult.ConvertIfNeeded(typeof(T));
 
-                if (validatedResult.Result == ArgumentConversionResultType.Successful
-                    && convertedResult.Result == ArgumentConversionResultType.NoArgument)
+                if (validatedResult is { Result: ArgumentConversionResultType.Successful }
```

Comment:
do you find this syntax more readable?

Reply:
I prefer it as a best practice because the behavior of `==` can be changed by operator overloading.